src/Models/CommunityModel.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+export type CommunityModel = {
+  moduleId: string;
+  name: string;
+  instructor: Instructor;
+  subscribersCount: number;
+  rating: Rating;
+  joined: boolean;
+};
+
+type Instructor = {
+  id: number;
+  name: string;
+};
+
+type Rating = {
+  id: number;
+  teaching: number;
+  content: number;
+  workload: number;
+};
+
+export function parser(object: any): CommunityModel {
+  return {
+    moduleId: object.moduleId,
+    name: object.name,
+    instructor: object.instructor,
+    subscribersCount: object.subscribersCount,
+    rating: object.rating,
+    joined: object.joined,
+  };
+}
```

Comment:
The problem was with the type. Typescript had a problem as the type of the object was any and so I made this function so that the type is a CommunityModel. Solved my problem, but if there is an other solution I am happy to know.